### PR TITLE
Posts from contacts can now be collapsed

### PR DESCRIPTION
**Pull request by annando**
When a contact is now set to "collapse", then all their postings are collapsed by default.

This is helpful when you often but not always want to ignore content by some contact.

Edit: Like @AlfredSK pointed out: This is exactly what the addon "blockem" is doing, thus it will make this addon obsolete.

See also PR https://github.com/friendica/friendica-addons/pull/1347

### CODE REVIEW

**Comment by AlfredSK** [comment]
Well, that's what the add-on Blockem is doing. Right?

**Comment by annando** [comment]
Yeah, exactly. Thanks for the hint.

File: src/Module/Contact.php [hunk 1]
```diff
@@ -382,11 +412,12 @@ protected function content(array $request = []): string
...
-			case 'pending':	 $header .= ' - ' . DI::l10n()->t('Pending'); break;
-			case 'blocked':	 $header .= ' - ' . DI::l10n()->t('Blocked'); break;
-			case 'hidden':   $header .= ' - ' . DI::l10n()->t('Hidden'); break;
-			case 'ignored':  $header .= ' - ' . DI::l10n()->t('Ignored'); break;
-			case 'archived': $header .= ' - ' . DI::l10n()->t('Archived'); break;
+			case 'pending':	  $header .= ' - ' . DI::l10n()->t('Pending'); break;
+			case 'blocked':	  $header .= ' - ' . DI::l10n()->t('Blocked'); break;
+			case 'hidden':    $header .= ' - ' . DI::l10n()->t('Hidden'); break;
+			case 'ignored':   $header .= ' - ' . DI::l10n()->t('Ignored'); break;
+			case 'collapsed': $header .= ' - ' . DI::l10n()->t('collapsed'); break;
```
Comment (MrPetovan) [inline src/Module/Contact.php]:
```suggestion
			case 'collapsed': $header .= ' - ' . DI::l10n()->t('Collapsed'); break;
```

File: src/Model/Item.php [hunk 2]
```diff
@@ -3048,6 +3048,9 @@ public static function prepareBody(array &$item, bool $attach = false, bool $is_
 		// Compile eventual content filter reasons
 		$filter_reasons = [];
 		if (!$is_preview && DI::userSession()->getPublicContactId() != $item['author-id']) {
+			if (Contact\User::isCollapsed($item['author-id'], $item['uid'])) {
+				$filter_reasons[] = DI::l10n()->t('Content from %s is collapsed', $item['author-name']);
+			}
```
Comment (MrPetovan) [inline src/Model/Item.php:3053]:
```suggestion
			}

```

File: src/Module/Contact/Profile.php [hunk 3]
```diff
@@ -197,7 +197,23 @@ protected function content(array $request = []): string
...
+				// @TODO: add $this->localRelationship->save($localRelationship);
+				DI::sysmsg()->addInfo($message);
+			}
+
+			if ($cmd === 'collapse') {
+				if ($localRelationship->collapsed) {
+					// @TODO Backward compatibility, replace with $localRelationship->unblock()
+					Contact\User::setCollapsed($contact['id'], DI::userSession()->getLocalUserId(), false);
 
+					$message = $this->t('Contact has been collapsed');
```
Comment (MrPetovan) [inline src/Module/Contact/Profile.php]:
```suggestion
					$message = $this->t('Contact has been uncollapsed');
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Please update the `messages.po` file after your last changes.

**Pull request merged**